lib/HttpClient.js

```diff
@@ -368,6 +368,7 @@ function rawRequest(opts, cb) {
         if (_socket.writable && !_socket._connecting) {
             startRequestTimeout();
```

Comment:
is there a possibility to start a request timeout more than once?

Reply:
Can req's `socket` event fire more than once?

Reply:
I don't think so, but we call `startRequestTimeout()` on the socket's `'connect'` event at https://github.com/restify/clients/pull/187/files#diff-dff98cc089bea71936ad27afa4770848R395. It's not clear to me whether there's a case where both of those call sites can be called for a single request, and whether that would be a problem. It just stuck me as odd.

Reply:
It looks like if both timeouts were triggered, it should blow up since `emitResult` is once.strict'ed. Not that that makes it less suspicious.